src/app/card-search.tsx

```diff
@@ -60,7 +60,11 @@ export const CardSearch: FC<CardSearchProps> = ({ initialQuery = '' }) => {
     startTransition(async () => {
       const cardData = await getCards(query, page);
       if (cardData.length > 0) {
-        setSearchResults((searchResults) => [...searchResults, ...cardData]);
+        setSearchResults((searchResults) => {
+          const newPages = searchResults.slice(0);
```

Comment:
Is there anything that prevents the `cardData` to be coming from an old `query` that no longer matches the current `searchQuery` after a `resetSearch()`?

Reply:
No, there isn't anything to prevent that scenario. This is actually a bug in the current implementation. 

To reproduce the bug, the steps are:
1. Start typing the name of a card
2. Wait for the loading state to appear
3. Delete the card name
4. Observe the results of the query appear

Here's a video that demonstrates the scenario with the card [Bria, Riptide Rogue](https://scryfall.com/card/blb/379/bria-riptide-rogue):

![card-spy-bug-reset-search](https://github.com/user-attachments/assets/99ecba7f-8aa6-4986-ba09-89e0bce36ddf)